Improved sandbox script to delete appropriately

pbking opened the pull request:
This change collects a list of what files have been deleted in the
repository and removes them from the sandbox via rm.

If those files have since returned (deleted but returned) they will
still be deleted but, since those files will be in your local copy will
be synced to the sandbox.

Also this change is determining which files that have been deleted locally but NOT as a committed change to the repository and removing those files from the sandbox.

pbking committed:
Improved sandbox script to delete appropriately

This change collects a list of what files have been deleted in the
repository and removes them from the sandbox via rm.

If those files have since returned (deleted but returned) they will
still be deleted but, since those files will be in your local copy will
be synced to the sandbox.

MaggieCabrera commented on sandbox.sh:
```diff
@@ -125,6 +125,13 @@ How do you wish to proceed? [1]"
 
   fi
 
+  # Determine which files have been removed from the repository (as of this head, within the past month) according to git
+  files_to_delete=$(git log --format=format:"" --name-only --since="last month" --diff-filter=D HEAD)
+
+  # Remove all of those files from the destination.  (Note, if a file has since been re-added to the repo it will be uploaded in the next step)
+  rmcmd="ssh $SANDBOX_USER@$SANDBOX_LOCATION 'cd $SANDBOX_PUBLIC_THEMES_FOLDER && rm -f $files_to_delete'"
+  eval $rmcmd;
+
```
I know you are not done but this worked a treat to fix the issues I had yesterday. Thank you

pbking committed:
Adjusted script to remove any files deleted local

With this change any files that have been DELETED locally, but not
comitted as a change, will be removed from the sandbox when executing
the sandbox sync operation.

pbking committed:
Determining deleted files accounts for moved

With this change the logic that determines which files should be flagged
for deletion takes into account file that git has flagg as "renamed".

jffng committed:
Fix spelling mistake :)

jffng reviewed (approved):
I pushed a spelling change but this works in my testing!

What I did to test: 

- Removed a file tracked by git and svn
- I did _not_ commit that deletion
- Ran `./sandbox.sh push` and went with option 2
- Verified the file was removed on my sandbox
